packages/openapi-generator/src/generator.ts

```diff
@@ -170,11 +148,84 @@ function parseServiceName(filePath: string): string {
   return parse(filePath).name.replace(/-openapi$/, '');
 }
 
+/**
+ * Generates an OpenAPI Service from a file.
+ * @param filePath The filepath where the service to generate is located.
+ * @param options  Options to configure generation.
+ * @param vdmMapping The vdmMapping for the OpenAPI generation.
+ * @param uniqueNameGenerator The uniqueNameGenerator with a kebab seperator.
+ */
+async function generateFromFile(
+  filePath: string,
+  options: GeneratorOptions,
+  vdmMapping: VdmMapping,
+  uniqueNameGenerator: UniqueNameGenerator
+): Promise<void> {
+  const serviceName = uniqueNameGenerator.generateAndSaveUniqueName(
+    parseServiceName(filePath)
+  );
+  const serviceDir = resolve(options.outputDir, serviceName);
+
+  let openApiDocument;
+  try {
+    openApiDocument = await convertOpenApiSpec(filePath);
+  } catch (err) {
+    logger.error(
+      `Could not convert document at ${filePath} to the format needed for parsing and generation. Skipping service generation.`
+    );
+    return;
+  }
+  const convertedInputFilePath = resolve(serviceDir, 'open-api.json');
+  const parsedOpenApiDocument = await parseOpenApiDocument(
+    openApiDocument,
+    serviceName,
+    filePath,
+    vdmMapping
+  );
+
+  if (!parsedOpenApiDocument.operations.length) {
+    logger.warn(
+      `The given OpenApi specificaton does not contain any operations. Skipping generation for input file: ${filePath}`
+    );
+    return;
+  }
+
+  await mkdir(serviceDir, { recursive: true });
+  await writeFile(
+    convertedInputFilePath,
+    JSON.stringify(openApiDocument, null, 2)
+  );
+
+  await generateOpenApiService(convertedInputFilePath, serviceDir);
+  await generateSDKSources(serviceDir, parsedOpenApiDocument, options);
+}
+
+/**
+ * Recursively searches through a given input path and returns all file paths as a string array.
+ * @param input the path to the input directory.
+ * @returns all file paths as a string array.
+ */
+async function getInputFilePaths(input: string): Promise<string[]> {
+  const directoryContents = await readdir(input, { withFileTypes: true });
+  const inputFilePaths = directoryContents.reduce(
+    async (paths: Promise<string[]>, directoryContent) => {
+      const contentPath = resolve(input, directoryContent.name);
+      const newInputFilePaths = directoryContent.isDirectory()
+        ? await getInputFilePaths(contentPath)
+        : [contentPath];
+      return [...(await paths), ...newInputFilePaths];
+    },
+    Promise.resolve([])
+  );
+  return inputFilePaths;
+}
+
 /**
  * Get the current SDK version from the package json.
  * @returns The SDK version.
  */
-export function getSDKVersion(): string {
-  return JSON.parse(readFileSync(resolve(__dirname, '../package.json'), 'utf8'))
-    .version;
+export async function getSDKVersion(): Promise<string> {
```

Comment:
I know you didn't implement this, but this is not following our coding conventions - it should be `getSdkVersion`. Could you please fix that?